common/src/Model/Brush.cpp

```diff
@@ -947,8 +970,23 @@ namespace TrenchBroom {
             assert(remaining.vertexCount() + moving.vertexCount() == vertexCount());
             
             // Will vertices be removed?
-            if (!allowVertexRemoval && result.vertexCount() < m_geometry->vertexCount())
-                return false;
+            if (!allowVertexRemoval) {
+                if (result.vertexCount() < m_geometry->vertexCount())
```

Comment:
> Actually check that all of the moving verts were not deleted. Comparing the vertex counts is not good enough, because moving a face can cause other faces to merge and eliminate verts; BrushTest.movePolygonRemainingPolyhedron tests this.

I can't think of a situation where moving a face may cause a vertex to be removed, and still be valid. It may cause other faces to be merged, but not vertices to be removed. Such cases must be denied.